conanfile.py

```diff
@@ -20,10 +20,15 @@ class OdbcConan(ConanFile):
 
     def configure(self):
         del self.settings.compiler.libcxx  # Pure C
+
+    def config_options(self):
         if self.settings.os == "Windows":
+            del self.options.shared
             del self.options.fPIC
 
     def source(self):
+        if self.settings.os == 'Windows':
```

Comment:
not sure about this one. conditional source method wasn't recommended by conan developers, as it breaks caching.
e.g if you run two commands:
```
conan install -s os=Windows ...
conan install -s os=Linux ...
```
the second will fail, because of empty source cached by the first command.
@Croydon do you think if it's safe to make an exception for odbc?

Reply:
You are right, I had forgotten, but I hit this problem with Qt recipe a long time ago, where sources downloaded depended on the options.